cmd/osmosisd/cmd/forceprune.go

```diff
@@ -69,92 +69,122 @@ func forceprune() *cobra.Command {
-			prunedBlocks, err := bs.PruneBlocks(current_height - full_height)
-			defer db_bs.Close()
+			err = compactBlockStore(db_path)
 			if err != nil {
 				return err
 			}
-			fmt.Println("Pruned Block Store ...", prunedBlocks)
-			db_bs.Close()
 
-			fmt.Println("Compacting Block Store ...")
+			return forcepruneStateStore(db_path, startHeight, currentHeight, min_height, full_height)
+		},
+	}
 
-			db, err := leveldb.OpenFile(db_path+"/blockstore.db", &opts)
-			defer db.Close()
-			if err != nil {
-				return err
-			}
-			if err = db.CompactRange(*util.BytesPrefix([]byte{})); err != nil {
-				return err
-			}
+	cmd.Flags().StringP(fullHeight, "f", defaultFullHeight, "Full height to chop to")
+	cmd.Flags().StringP(minHeight, "m", defaultMinHeight, "Min height for ABCI to chop to")
+	return cmd
+}
 
-			db, err = leveldb.OpenFile(db_path+"/state.db", &opts)
-			if err != nil {
-				return err
-			}
-			stateDBKeys := []string{kValidators, kConsensusParams, kABCIResponses}
-			fmt.Println("Pruning State Store ...")
-			for i, s := range stateDBKeys {
-				fmt.Println(i, s)
-
-				retain_height := int64(0)
-				if s == kABCIResponses {
-					retain_height = current_height - min_height
-				} else {
-					retain_height = current_height - full_height
-				}
+func pruneBlockStoreAndGetHeights(dbPath string, fullHeight int64) (
+	startHeight int64, currentHeight int64, err error,
+) {
+	opts := opt.Options{
+		DisableSeeksCompaction: true,
+	}
 
-				batch := new(leveldb.Batch)
-				curBatchSize := uint64(0)
+	db_bs, err := tmdb.NewGoLevelDBWithOpts("blockstore", dbPath, &opts)
+	if err != nil {
+		return 0, 0, err
+	}
+	defer db_bs.Close()
 
-				fmt.Println(start_height, current_height, retain_height)
+	bs := tmstore.NewBlockStore(db_bs)
+	startHeight = bs.Base()
+	currentHeight = bs.Height()
 
-				for c := start_height; c < retain_height; c++ {
-					batch.Delete([]byte(s + strconv.FormatInt(c, 10)))
-					curBatchSize++
+	fmt.Println("Pruning Block Store ...")
+	prunedBlocks, err := bs.PruneBlocks(currentHeight - fullHeight)
+	if err != nil {
+		return 0, 0, err
+	}
+	fmt.Println("Pruned Block Store ...", prunedBlocks)
+	return startHeight, currentHeight, nil
+}
 
-					if curBatchSize%batchMaxSize == 0 && curBatchSize > 0 {
-						err := db.Write(batch, nil)
-						if err != nil {
-							return err
-						}
-						batch.Reset()
-						batch = new(leveldb.Batch)
-					}
-				}
+func compactBlockStore(dbPath string) error {
+	compactOpts := opt.Options{
+		DisableSeeksCompaction: true,
+	}
+
+	fmt.Println("Compacting Block Store ...")
+
+	db, err := leveldb.OpenFile(dbPath+"/blockstore.db", &compactOpts)
+	defer db.Close()
+	if err != nil {
+		return err
+	}
+	if err = db.CompactRange(*util.BytesPrefix([]byte{})); err != nil {
+		return err
+	}
+	return nil
+}
+
+func forcepruneStateStore(dbPath string, startHeight, currentHeight, minHeight, fullHeight int64) error {
+	opts := opt.Options{
+		DisableSeeksCompaction: true,
+	}
```

Comment:
I chose to duplicate this in each function, since its pretty unclear to me that longer term, these won't be different for every function